### PR TITLE
Support `arrowGlacier` hardfork

### PR DESCRIPTION
**Pull request by feuGeneA**

### CODE REVIEW

File: packages/hardhat-core/src/internal/core/providers/accounts.ts [hunk 1]
```diff
@@ -253,6 +253,7 @@ export class LocalAccountsProvider extends ProviderWrapperWithChainId {
       gasLimit: transactionRequest.gas,
     };
 
+    // TODO: consider changing instances of "london" below to ["latest hardfork"]
```
Comment (alcuadrado) [inline packages/hardhat-core/src/internal/core/providers/accounts.ts:256]: London here is just a way to signal to ethereumjs that EIP-1559 may be used

File: packages/hardhat-core/src/internal/hardhat-network/provider/fork/ForkBlockchain.ts [hunk 2]
```diff
@@ -272,7 +272,7 @@ export class ForkBlockchain implements HardhatBlockchainInterface {
     // one for London.
     const common = this._common.copy();
     if (rpcBlock.baseFeePerGas !== undefined) {
-      common.setHardfork("london");
+      common.setHardfork("london"); // TODO: consider changing this to "latest hardfork"
```
Comment (alcuadrado) [inline packages/hardhat-core/src/internal/hardhat-network/provider/fork/ForkBlockchain.ts:275]: Same here

File: packages/hardhat-core/src/internal/hardhat-network/provider/transactions/ReadOnlyValidEIP1559Transaction.ts [hunk 3]
```diff
@@ -57,7 +57,7 @@ export class ReadOnlyValidEIP1559Transaction extends FeeMarketEIP1559Transaction
   constructor(sender: Address, data: FeeMarketEIP1559TxData = {}) {
     const fakeCommon = new Common({
       chain: "mainnet",
-      hardfork: "london",
+      hardfork: "london", // TODO: consider chaning this to ["latest hardfork"]
```
Comment (alcuadrado) [inline packages/hardhat-core/src/internal/hardhat-network/provider/transactions/ReadOnlyValidEIP1559Transaction.ts:60]: Same

File: packages/hardhat-core/test/internal/hardhat-network/stack-traces/execution.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ const senderPrivateKey = Buffer.from(
 const senderAddress = privateToAddress(senderPrivateKey);
 
 export async function instantiateVm(): Promise<VM> {
-  const account = Account.fromAccountData({ balance: 1e18 });
+  const account = Account.fromAccountData({ balance: 1e15 });
```
Comment (alcuadrado) [inline packages/hardhat-core/test/internal/hardhat-network/stack-traces/execution.ts:16]: Was this actually necessary?

**Comment by alcuadrado** [comment]
Reviewing the tests is a bit complicated, I think GH gets somewhat lost with the diff. Did you just make the tests that used london use london AND arrow glacier now? That sounds clever

File: packages/hardhat-core/src/internal/constants.ts [hunk 5]
```diff
@@ -18,6 +18,7 @@ export const HARDHAT_NETWORK_SUPPORTED_HARDFORKS = [
   "muirGlacier",
   "berlin",
   "london",
+  "arrowGlacier",
```
Comment (fvictorio) [inline packages/hardhat-core/src/internal/constants.ts:21]: We need to update the docs too.

File: packages/hardhat-core/test/internal/hardhat-network/provider/baseFeePerGas.ts [hunk 6]
```diff
@@ -122,39 +122,41 @@ describe("Block's baseFeePerGas", function () {
...
-                  "eth_getBlockByNumber",
-                  ["latest", false]
-                );
-
-                const latestBlockData = rpcToBlockData({
-                  ...latestBlockRpc,
-                  transactions: [],
-                });
+              for (const hardfork of ["london", "arrowGlacier"]) {
+                it(`should compute the next base fee correctly when ${hardfork} is activate`, async function () {
```
Comment (fvictorio) [inline packages/hardhat-core/test/internal/hardhat-network/provider/baseFeePerGas.ts:126]:
```suggestion
                it(`should compute the next base fee correctly when ${hardfork} is activated`, async function () {
```

File: packages/hardhat-core/src/internal/core/providers/accounts.ts (hunk 1 above)
Comment (feuGeneA) [reply packages/hardhat-core/src/internal/core/providers/accounts.ts:256]: On a call, Pato said this is fine as-is, TODO and all.

File: packages/hardhat-core/src/internal/hardhat-network/provider/fork/ForkBlockchain.ts (hunk 2 above)
Comment (feuGeneA) [reply packages/hardhat-core/src/internal/hardhat-network/provider/fork/ForkBlockchain.ts:275]: On a call, Pato said this is fine as-is, TODO and all.

File: packages/hardhat-core/test/internal/hardhat-network/stack-traces/execution.ts (hunk 4 above)
Comment (feuGeneA) [reply packages/hardhat-core/test/internal/hardhat-network/stack-traces/execution.ts:16]: We discussed this on a call. Yes, it is necessary, as exhibited by the CI failure for [this commit](https://github.com/nomiclabs/hardhat/pull/2124/commits/b037daff1ff2815758f526974591db813ad89558), and as explained in the commit message for [this other commit](https://github.com/nomiclabs/hardhat/pull/2124/commits/18a770ac82a340bc65c27d27d7e7adea977bb217).

File: packages/hardhat-core/src/internal/hardhat-network/provider/transactions/ReadOnlyValidEIP1559Transaction.ts (hunk 3 above)
Comment (feuGeneA) [reply packages/hardhat-core/src/internal/hardhat-network/provider/transactions/ReadOnlyValidEIP1559Transaction.ts:60]: On a call, Pato said this is fine as-is, TODO and all.

File: packages/hardhat-core/src/internal/constants.ts (hunk 5 above)
Comment (feuGeneA) [reply packages/hardhat-core/src/internal/constants.ts:21]: Addressed in https://github.com/nomiclabs/hardhat/pull/2124/commits/4ceff2dcffc9a254b5d8abde976f1a87b63b2d49

**Review by fvictorio — APPROVED** [review]

**Pull request merged**